Mage.Sets/src/mage/cards/h/HorrificAssault.java

```diff
@@ -0,0 +1,56 @@
+package mage.cards.h;
+
+import java.util.UUID;
+
+import mage.abilities.condition.common.PermanentsOnTheBattlefieldCondition;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.common.DamageWithPowerFromOneToAnotherTargetEffect;
+import mage.abilities.effects.common.GainLifeEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.TargetController;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreatureOrPlaneswalkerPermanent;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+
+/**
+ *
+ * @author grimreap124
+ */
+public final class HorrificAssault extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterCreatureOrPlaneswalkerPermanent(
+            "creature or planeswalker you don't control");
+
+    private static final FilterPermanent eldraziFilter = new FilterPermanent("you control an Eldrazi");
```

Comment:
`new FilterControlledPermanent(SubType.ELDRAZI, "you control an Eldrazi")` would prevent the need to alter the filter by adding predicates.